src/crustls.h

```diff
@@ -374,6 +385,20 @@ typedef const struct rustls_certified_key *(*rustls_client_hello_callback)(rustl
  */
 size_t rustls_version(char *buf, size_t len);
 
+/**
+ * Return a 16-bit unsigned integer corresponding to this cipher suite's assignment from
+ * <https://www.iana.org/assignments/tls-parameters/tls-parameters.xhtml#tls-parameters-4>.
+ * The bytes from the assignment are interpreted in network order.
+ */
+uint16_t rustls_supported_ciphersuite_get_suite(const struct rustls_supported_ciphersuite *supported_ciphersuite);
+
+/**
+ * Get a pointer to a member of rustls' list of supported cipher suites. This will return non-NULL
+ * for 0 <= i < RUSTLS_ALL_CIPHERSUITES_LEN. The returned pointer is valid for the lifetime of
+ * the program and may be used directly when building a ClientConfig or ServerConfig.
+ */
+const struct rustls_supported_ciphersuite *rustls_all_ciphersuites_get(size_t i);
```

Comment:
For applications using a previous higher number from a `#define RUSTLS_ALL_CIPHERSUITES_LEN`, this call will return NULL unexpectedly on a new crustls version with less supported ciphers.

Reply:
It would be better to have something like this:

```
rustls_supported_ciphersuite *suite;
int i = 0;
while ((cipher = rustls_supported_ciphersuite_get(i++)) {
   /* remember cipher */
}
```

Reply:
Ooh, this is a good pattern. Thanks for the suggestion! And good point about ABI compatibility.